### ISSUE TITLE
'URI.escape is obsolete' error appears repeatedly.

### ISSUE THREAD

**Issue by danielbayley** [bundler/bundler#3888]
`…/Ruby/gems/bundler-1.10.6/lib/bundler/fetcher/dependency.rb:70:in 'dependency_api_uri': warning: URI.escape is obsolete`

OS X 10.10.4

```bash
$ gem env
RubyGems Environment:
  - RUBYGEMS VERSION: 2.0.14
  - RUBY VERSION: 2.0.0 (2014-05-08 patchlevel 481) [universal.x86_64-darwin14]
  - INSTALLATION DIRECTORY: /Users/Dan/Library/Scripts/Libraries/Ruby
  - RUBY EXECUTABLE: /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
  - EXECUTABLE DIRECTORY: /Users/Dan/Library/Scripts/Libraries/Ruby/bin
  - RUBYGEMS PLATFORMS:
    - ruby
    - universal-darwin-14
  - GEM PATHS:
     - /Users/Dan/Library/Scripts/Libraries/Ruby
     - /Users/Dan/.gem/ruby/2.0.0
     - /Library/Ruby/Gems/2.0.0
     - /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/gems/2.0.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
```

**Comment by EduardoBautista** [comment]
Here is the PR: https://github.com/bundler/bundler/pull/3892

**Comment by segiddins** [comment]
FWIW, I don't see that message running 2.2.2

**Comment by indirect** [comment]
It looks like ruby-core "un-obsoleted" it later, but we investigated together and it's actually a bug to use URI.escape—we should be using CGI.escape instead.

**Comment by segiddins** [comment]
We're using `URI.encode`, not `URI.escape` ?

**Comment by indirect** [comment]
Those are aliases ¯\_(ツ)_/¯

**Comment by danielbayley** [comment]
So will this permanently affect Ruby 2.0.0 (the default OS X installation)?

**Comment by indirect** [comment]
@danielbayley no, there's already a pull request open to fix it: #3892.
